src/ResourceTile/ResourceTile.tsx

```diff
@@ -0,0 +1,244 @@
+import * as classnames from "classnames";
+import { Button, FlexBox, ItemAlign, Justify, Tooltip } from "../index";
+import { DragHandleIcon } from "./DragHandleIcon";
+import * as _ from "lodash";
+import * as React from "react";
+import * as FontAwesome from "react-fontawesome";
+
+import "./ResourceTile.less";
+
+const TITLE_MAX_LENGTH = 30;
+
+export type IconSize = "small" | "medium" | "large";
+export type IconOrientation = "landscape" | "portrait" | "unknown";
+
+export const CssClasses = {
+  ACTION: "ResourceTile--action",
+  ACTION_BUTTON: "ResourceTile--actionButton",
+  CONTAINER: "ResourceTile",
+  DRAGGABLE_ICON: "ResourceTile--draggableIcon",
+  LINK: "ResourceTile--link",
+  ICON_CONTAINER: "ResourceTile--iconContainer",
+  ICON: "ResourceTile--icon",
+  OVERLAYS: "ResourceTile--overlays",
+  OVERLAY: "ResourceTile--overlay",
+  TITLE_CONTAINER: "ResourceTile--titleContainer",
+  TITLE_AND_NOTES: "ResourceTile--titleAndNotes",
+  TITLE: "ResourceTile--title",
+  NOTIFICATION: "ResourceTile--notification",
+  NOTES: "ResourceTile--notes",
+  ADDITIONAL_INFO: "ResourceTile--additionalInfo",
+};
+
+const SizeCssClasses: Record<IconSize, string> = {
+  small: "ResourceTile--size--small",
+  medium: "ResourceTile--size--medium",
+  large: "ResourceTile--size--large",
+};
+
+const IconOrientationCssClasses: Record<IconOrientation, string> = {
+  landscape: "ResourceTile--icon--landscape",
+  portrait: "ResourceTile--icon--portrait",
+  unknown: "",
+};
+
+export interface Action {
+  content: React.ReactNode;
+  href?: string;
+  onClick?: () => void;
+}
+
+export interface Props {
+  actions: [] | [Action] | [Action, Action];
```

Comment:
This __may__ be an example of trying to be too strict with types so I'm open to opinions on this! Previously, we represented this as `Action[]` but always only used the first two actions in the array. Instead, I'm expressing this implicit behavior in an explicit way by  changing the type to only allow tuples of Actions with up to 2 entries. This may cause it to be slightly harder to pass in actions to this component since you have to write it in a way that convinces the compiler you're only passing in at most two actions.

Reply:
This seems totally reasonable to me!